cmd/query/app/server.go

```diff
@@ -274,10 +274,14 @@ func (s *Server) initListener() (cmux.CMux, error) {
 func (s *Server) Start() error {
 	cmuxServer, err := s.initListener()
 	if err != nil {
+		s.logger.Error("Failed to initialize listener", zap.Error(err))
 		return err
```

Comment:
there is no need to log, just wrap the error
```suggestion
		return fmt.Errorf("query server failed to initialize listener: %w", err)
```